app/src/main/java/zapsolutions/zap/HomeActivity.java

```diff
@@ -406,4 +428,74 @@ public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, Strin
             }
         }
     }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+
+        PendingIntent pendingIntent = PendingIntent.getActivity(
+                this, 0, new Intent(this, getClass()).addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP), 0);
+        if (mNfcAdapter != null) {
+            mNfcAdapter.enableForegroundDispatch(this, pendingIntent, NfcUtil.IntentFilters(), null);
+        }
+    }
+
+    @Override
+    protected void onPause() {
+        super.onPause();
+        if (mNfcAdapter != null) {
+            mNfcAdapter.disableForegroundDispatch(this);
+        }
+    }
+
+    @Override
+    protected void onNewIntent(Intent intent) {
+        super.onNewIntent(intent);
+        NfcUtil.readTag(this, intent, new NfcUtil.OnNfcResponseListener() {
+            @Override
+            public void onSuccess(String payload) {
+                readInvoice(payload);
+            }
+        });
+    }
+
+    private void readInvoice(String invoice) {
+        InvoiceUtil.readInvoice(HomeActivity.this, compositeDisposable, invoice, new InvoiceUtil.OnReadInvoiceCompletedListener() {
+            @Override
+            public void onValidLightningInvoice(PayReq paymentRequest, String invoice) {
+                Intent intent = new Intent();
+                intent.putExtra("onChain", false);
+                intent.putExtra("lnPaymentRequest", paymentRequest.toByteArray());
+                intent.putExtra("lnInvoice", invoice);
+                SendBSDFragment sendBottomSheetDialog = new SendBSDFragment();
+                sendBottomSheetDialog.setArguments(intent.getExtras());
+                sendBottomSheetDialog.show(getSupportFragmentManager(), "sendBottomSheetDialog");
+            }
+
+            @Override
+            public void onValidBitcoinInvoice(String address, long amount, String message) {
+                Intent intent = new Intent();
+                intent.putExtra("onChain", true);
+                intent.putExtra("onChainAddress", address);
+                intent.putExtra("onChainAmount", amount);
+                intent.putExtra("onChainMessage", message);
+                SendBSDFragment sendBottomSheetDialog = new SendBSDFragment();
+                sendBottomSheetDialog.setArguments(intent.getExtras());
+                sendBottomSheetDialog.show(getSupportFragmentManager(), "sendBottomSheetDialog");
+            }
+
+            @Override
+            public void onError(String error, int duration) {
+                showError(error, duration);
+            }
+        });
+    }
+
+    private void showError(String message, int duration) {
```

Comment:
Would it make sense to move this method to `BaseAppCompatActivity` and reused in other activities/fragments?

Reply:
good remark, I will change that!